Add new runtime setting for JSON string used with mapping (#7378)
mendix/docs: content/en/docs/refguide/modeling/integration/mapping-documents/import-mappings.md

To define a parameter for your mapping, click the parameter box and select the data type. You can also drag an entity into the parameter box. 

You can use a parameter as a key or in a microflow to obtain objects. When you use an entity parameter, you can set associations to it. When you use a primitive parameter (string, integer, etc.), you can write the value to an attribute of an object that is being imported.

## 4 Troubleshooting{#troubleshooting}

If you are importing a very long JSON string, the underlying conversion library may not be able to handle it. In this case you will get a message similar to the following:

> `Caused by: com.fasterxml.jackson.databind.JsonMappingException: String length (20051119) exceeds the maximum length (20000000)`

Reviewer: This is the actual exception the customer receiving based on the ticket `com.mendix.systemvideinterfaces.MendixRuntimeException: com.fasterxm1.jackson.core.exc.StreamConstraintsException: String value length (20051112) exceeds the maximum allowed (20000000, from 'StreamReadConstraints.getMaxStringLength()')`
Would it help the user better if the exception as close as possible?

Reviewer 2: Yes, I'll put the actual exception in. Thanks.
```suggestion
`com.mendix.systemvideinterfaces.MendixRuntimeException: com.fasterxm1.jackson.core.exc.StreamConstraintsException: String value length (20051112) exceeds the maximum allowed (20000000, from 'StreamReadConstraints.getMaxStringLength()')`
```

Mendix uses the Jackson Core XML library when performing the import mapping and this [has a limit of 20 million](https://javadoc.io/static/com.fasterxml.jackson.core/jackson-core/2.15.1/com/fasterxml/jackson/core/StreamReadConstraints.html#DEFAULT_MAX_STRING_LEN) characters (in earlier versions 5 million) in the JSON string.

In Mendix version 10.9.0 and above, you can override this using the [mapping.import.MaxJsonReadingLength](/refguide/custom-settings/#mapping.import.MaxJsonReadingLength) custom setting.

> Mendix recommends that you only add this setting if you will be getting very long JSON strings as it will have an impact on performance.

Reviewer: Rephrase to "...may result in more memory usages." I think this would give more information rather than performance.

Reviewer 2: Thanks for clarifying - I think we should be as accurate as possible.
```suggestion
Mendix recommends that you only add this setting if you will be getting very long JSON strings as it may result in more memory usage.
```
